guillotina/tests/test_content.py

```diff
@@ -9,13 +9,34 @@
 from guillotina.exceptions import NoPermissionToAdd
 from guillotina.exceptions import NotAllowedContentType
 from guillotina.interfaces import IApplication
+from guillotina.interfaces import IItem
 from guillotina.interfaces.types import IConstrainTypes
+from guillotina.schema import Dict
+from guillotina.schema import TextLine
 from guillotina.tests import utils
 
 import pickle
 import pytest
 
 
+class ICustomContentType(IItem):
+
+    images = Dict(
+        key_type=TextLine(),
+        value_type=TextLine(),
+        required=False,
+        default={}
```

Comment:
Should be with factory, otherwise is the global object and the test does not makes sense

Reply:
I ran the tests without the fix and I could reproduce the bug (that's because `_default_from_schema()` creates a deepcopy of the default value: https://github.com/plone/guillotina/blob/master/guillotina/content.py#L138). I think this implementation details should be documented to know when to use `default` and `defaultFactory`. That said, I have no problem to change this :)